src/fabric.c

```diff
@@ -875,6 +894,7 @@ static int ofi_layering_ok(const struct fi_provider *provider,
 			   uint64_t flags)
 {
 	char *prov_name;
+	struct ofi_prov *core_ofi_prov;
```

Comment:
The comment above this function needs to be updated.  See rule 1d, which should be reworded to something like, "A utility provider will not layer over a 'non-layering' core provider unless explicitly requested."

Yeah, I'm not sure what to call these marked providers other than non-layering.